.github/workflows/build.yml

```diff
@@ -1,399 +1,104 @@
 ---
 name: CI
 
-on: [push, pull_request]
```

Comment:
I wouldn't delete this as we would like to know before making a PR whether our code is passing CI or not

Reply:
"before making a PR"


You're going to have to educate me.  You mean that "push" tells you if your code in main would pass?? pull_request if your PR passes?

Reply:
That's not a problem at all! Yes, the `push` key means that this workflow will run on the state of `push` & `pull_request` of the branch.